tango_ros_common/tango_ros_native/src/occupancy_grid_file_io.cpp

```diff
@@ -27,9 +29,47 @@
 namespace {
 void AddTrailingSlashToDirectoryPathIfNeeded(std::string& directory_path) {
   if (!directory_path.empty() && directory_path.back() != '/') {
-    directory_path += "/";
+    directory_path += '/';
   }
 }
+bool DirectoryPathExists(const std::string& path) {
+  struct stat st;
+  if (stat(path.c_str(), &st) == 0 && (st.st_mode & S_IFDIR)) {
+    return true;
+  } else {
+    return false;
+  }
+}
+int MakeDirectoryPath(const std::string& input_path, mode_t mode) {
+  if (input_path.empty()) {
+    // Path is empty, there is nothing to create so we return success.
+    return 0;
+  }
+
+  size_t previous_slash_pos = 0;
+  size_t current_slash_pos;
+  std::string dir;
+  int make_dir_status = 0;
+
+  std::string path = input_path;
+  CHECK(!path.empty());
+  AddTrailingSlashToDirectoryPathIfNeeded(path);
+
+  while ((current_slash_pos = path.find_first_of('/', previous_slash_pos)) !=
+         std::string::npos) {
+    dir = path.substr(0, current_slash_pos++);
+    previous_slash_pos = current_slash_pos;
+    if (dir.empty())
```

Comment:
Add curly brackets for multiline expression. Also combine if statement here and below with OR.